media/windows/calendar-reservations/calendar-reservations.py

```diff
@@ -114,15 +327,35 @@ def check_response_status(events, calendar, log):
 
 ####################################################################
 
-def respond_to_events(events, service, calendar, log):
+def check_for_conflicts(event_to_check, events, calendar, log):
+    # iterate through the list of all of the events and finds any that conflict with the event in question
+    eventConflicts = False
+
+    if(calendar["name"] != "Epiphany Events"): # will not check for conflicts on the Epiphany Events calendar
+        for event in events:
```

Comment:
Note that we know that the list of events we got back from Google is ordered.  Can we use that somehow to speed up the searching for conflicts?  (and/or does it really matter / does it materially speed up anything?)

If we're worried about the number of API calls and/or the time it takes to download the whole calendar (e.g., Epiphany Events), should we use the `orderBy=updated` option to find the events most recently updated?  

Stream of consciousness thinking here: we could keep a local cache of the most recently updated event that we've examined/cleared (e.g., in a local SQL database, or perhaps even a simple JSON file! -- although we would need to figure out how to handle if we a) cache event XYZ as the most recent we've cleared, and then b) someone deletes event XYZ).  That way, if the first event in `orderBy=updated` is the same one as in our cache, then we can know immediately that we don't need to check anything, because nothing has changed on the calendar since the last time we checked.  But if something *has* changed, we can potentially limit the number of events we have to download via `list()` because we'll know the exact time range of the events that have been updated / require us to reply.  Need to think this through a bit, but this could be a useful idea to noodle on...

Maybe that optimization can be for v2.0.  😉